File: eCommerce.Domain/Product/Product.cs
using System.Collections.Generic;
using eCommerce.Domain.Product.Enums;

namespace eCommerce.Domain.Product
{
	public class Product
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }
		public IEnumerable<string> Specifications { get; set; }
		public ProductCategory Category { get; set; }
		public string ImagePath { get; set; }
		public string MenuImagePath { get; set; }
		public bool IsRemove { get; set; }
	}
}

File: eCommerce.Site/ClientApp/src/domain/product/product.ts
import {ProductCategory} from "./productCategory";

export class Product{
    
    public id: number;
    public name: string;
    public description: string;
    public category: ProductCategory;
    public price: number;
    public imgUrl: string;
    
    constructor(
        
        id: number, 
        name: string,
        description: string,
        category: ProductCategory,
        price: number,
        imageUrl: string
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.category = category;
        this.price = price;
        this.imgUrl = imageUrl;
    }
    
    
    
}